app/models/concerns/twitter_user/builder.rb

```diff
@@ -0,0 +1,114 @@
+    def build_relations
+      uid_i = uid.to_i
+      search_query = "@#{screen_name}"
+
+      if ego_surfing?
+        candidates = [
+          {method: :friends_parallelly, args: [uid_i]},
+          {method: :followers_parallelly, args: [uid_i]},
+          {method: :user_timeline, args: [uid_i]}, # for replying
+          {method: :search, args: [search_query]}, # for replied
+          {method: :home_timeline, args: [uid_i]},
+          {method: :mentions_timeline, args: [uid_i]}, # for replied
+          {method: :favorites, args: [uid_i]} # for favoriting
+        ]
+        if without_friends
+          candidates = candidates.slice(2, candidates.size - 2)
+          _statuses, _search_results, _, _mentions, _favorites = client._fetch_parallelly(candidates)
+          _friends = _followers = []
+        else
+          _friends, _followers, _statuses, _search_results, _, _mentions, _favorites = client._fetch_parallelly(candidates)
+        end
+      else
+        candidates = [
+          {method: :friends_parallelly, args: [uid_i]},
+          {method: :followers_parallelly, args: [uid_i]},
+          {method: :user_timeline, args: [uid_i]},
+          {method: :search, args: [search_query]},
+          {method: :favorites, args: [uid_i]}
+        ]
+        if without_friends
+          candidates = candidates.slice(2, candidates.size - 2)
+          _statuses, _search_results, _favorites = client._fetch_parallelly(candidates)
+          _friends = _followers = []
+        else
+          _friends, _followers, _statuses, _search_results, _favorites = client._fetch_parallelly(candidates)
+        end
+        _mentions = []
+      end
+
+      # Not using uniq for mentions, search_results and favorites intentionally
+
+      client._fetch_parallelly([
+                                 {method: :replying, args: [uid_i]}
+                               ])
+
+      _friends.each do |f|
+        friends.build(uid: f.id,
+                      screen_name: f.screen_name,
+                      user_info: f.slice(*TwitterUser::PROFILE_SAVE_KEYS).to_json)
+      end
+
+      _followers.each do |f|
+        followers.build(uid: f.id,
+                        screen_name: f.screen_name,
+                        user_info: f.slice(*TwitterUser::PROFILE_SAVE_KEYS).to_json)
+      end
+
+      _statuses.each do |s|
+        statuses.build(uid: uid,
+                       screen_name: screen_name,
+                       status_info: s.slice(*Status::STATUS_SAVE_KEYS).to_json)
+      end
+
+      _mentions.each do |m|
+        mentions.build(uid: m.user.id,
+                       screen_name: m.user.screen_name,
+                       status_info: m.slice(*Status::STATUS_SAVE_KEYS).to_json)
+      end
+
+      _search_results.each do |sr|
+        search_results.build(uid: sr.user.id,
+                             screen_name: sr.user.screen_name,
+                             status_info: sr.slice(*Status::STATUS_SAVE_KEYS).to_json,
```

Comment:
__[rubocop]__
* Line is too long. [86/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)